extension.js

```diff
@@ -36,6 +36,10 @@ class ExtensionDBusInterface {
             return;
 
         Main.wm.skipNextEffect(current_window.get_compositor_private());
+        if (settings.get_boolean('window-maximize')) {
+            settings.set_boolean('window-maximize', false);
```

Comment:
Because of `.unmaximize()` call at the end of this function, window's `maximized-vertically` property will be changed to `false`. And that should trigger `unmaximize_window` callback (probably should be renamed to `handle_maximized_changed` or something like that), and it will set `window-maximize` to `false` anyway. So this line should be unnecessary too